File: src/calc_23.js
'use strict';

const TOTAL = 1e5;
const MAX_PEOPLE = 23;
let count = 0;

let start = Date.now();

for (let i = 0; i < TOTAL; i++) {
    let days = [];

    for (let j = 0; j < MAX_PEOPLE; j++) {
        let birthday = Math.floor(Math.random() * 365);

        if (days.includes(birthday)) {
            count++;
            break;
        }

        days.push(birthday);
    }
}

let end = (Date.now() - start) / 1000;

console.log(`Probability of ${count / TOTAL}`);
console.log(`The calculations took ${end} seconds`);

File: src/dart/calc_23.dart
import 'dart:math';

void main() {
  const TOTAL = 100000;
  const MAX_PEOPLE = 23;
  int count = 0;

  var random = Random();
  var start = DateTime.now();

  for (int i = 0; i < TOTAL; i++) {
    List<int> days = [];

    for (int j = 0; j < MAX_PEOPLE; j++) {
      int birthday = random.nextInt(365);

      if (days.contains(birthday)) {
        count++;
        break;
      }

      days.add(birthday);
    }
  }

  var end = DateTime.now().difference(start).inMicroseconds;

  print('Probability of ${count / TOTAL}');
  print('The calculations took $end microseconds');
}
